### PR TITLE
[JavaScript] arrays - replace about.md file with concept files

### PR DESCRIPTION
**Pull request by ErikSchierboom**
In [this issue](https://github.com/exercism/v3/issues/2293) we're describing an evolution of the specification of this repo.
These changes include, amongst others, replacing the contents of the `after.md` document with individual concept documents. There are two documents per concept:

1. An `about.md` file containing the description of the concept
1. A `links.json` file containing a set of useful links related to the concept. 

This looks as follows in the repo:

```
<track>
├── concepts
│   ├── <concept>
│   │   ├── about.md
│   │   └── links.json
│   └── ...
```

This PR applies this change to this exercise by: 
                    
1. Copying the existing `after.md` file's contents to the `about.md` file of each concept listed in the exercise's `concepts` section in the `config.json` file.
1. Extract the links from the `after.md` file and put them into the `links.json` file

Before merging this PR, please check that the contents of the concept documents makes sense.
This is especially true when the exercise unlocks multiple concepts, as then the concept documents should be updated to only contain the information relevant to that concept.

Note: to make reviewing easier, the PR has been split into three separate commits. Please squash this PR upon merging.

### CODE REVIEW

File: languages/javascript/concepts/arrays/about.md [hunk 1]
```diff
@@ -1 +1,11 @@
-TODO: add information on arrays concept
+In Javascript, an array is actually just a regular object where the elements are properties of that object. It includes the `length` property and also lots of [useful methods][array-docs] for traversing and mutating the array.
```
Comment (SleeplessByte) [inline languages/javascript/concepts/arrays/about.md:1]: Change wording and remove `just`.

**Review by SleeplessByte — COMMENTED** [review]
Missing a lot of generic information about arrays.

@ErikSchierboom how can I link to related concepts??

**Comment by ErikSchierboom** [comment]
> @ErikSchierboom how can I link to related concepts?

We haven't yet defined a spec for this. I was thinking about this earlier today. We'll come up with something!

**Comment by SleeplessByte** [comment]
Perfect! 

It's blocking for some of it for me; let me know if there is a temporary
(or permanent) solution for that. 

On 2020-10-20 11:06, Erik Schierboom wrote:

>> @ErikSchierboom [1] how can I link to related concepts?
> 
> We haven't yet defined a spec for this. I was thinking about this earlier today. We'll come up with something! 
> 
> --
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub [2], or unsubscribe [3].

 

Links:
------
[1] https://github.com/ErikSchierboom
[2] https://github.com/exercism/v3/pull/2416#issuecomment-712707154
[3]

**Review by SleeplessByte — APPROVED** [review]

**Pull request merged**